src/classes/Gifts.cls

```diff
@@ -35,26 +35,72 @@
 
 public with sharing class Gifts {
 
-    private Set<String> captureGroupIds;
-    private List<DataImport__c> dataImports;
+    private Set<String> capturableGroupIds;
+    @TestVisible private List<DataImport__c> dataImports;
+    @TestVisible private List<DataImport__c> giftsToUpdate = new List<DataImport__c>();
 
     public Gifts(List<DataImport__c> dataImports) {
         this.dataImports = dataImports;
-        setCaptureGroupIds(dataImports);
+        setCapturableGroupIds(dataImports);
     }
 
-    private void setCaptureGroupIds(List<DataImport__c> dataImports) {
-        captureGroupIds = new Set<String>();
-        for (DataImport__c dataImport : dataImports) {
-            captureGroupIds.add(dataImport.Payment_Elevate_Batch_ID__c);
+    public List<DataImport__c> authorizedGifts() {
+        List<DataImport__c> giftsToReturn = new List<DataImport__c>();
+        for (DataImport__c gift : dataImports) {
+            if (gift.Payment_Status__c == ElevateTransactionStatus.AUTHORIZED.name()) {
+                giftsToReturn.add(gift);
+            }
         }
+        return giftsToReturn;
+    }
+
+    public void updateRecords() {
+        update giftsToUpdate;
+    }
+
+    public Set<String> capturableGroupIds() {
+        return capturableGroupIds;
     }
 
-    public Set<String> captureGroupIds() {
-        return captureGroupIds;
+    public void authorizedToPaymentProcessingStatus() {
+        for (DataImport__c gift : authorizedGifts()) {
+            gift.Payment_Status__c = ElevateTransactionStatus.PAYMENTPROCESSING.name();
+            giftsToUpdate.add(gift);
+        }
+    }
+
+    public void failGiftsByCaptureIdWithReason(Map<String, String> failReasonByCaptureGroupId) {
```

Comment:
This and the following two methods seem like they should be on a CaptureGroups domain class.